sessionctx/stmtctx/stmtctx.go

```diff
@@ -136,6 +176,8 @@ type StatementContext struct {
 	}
 	// PrevAffectedRows is the affected-rows value(DDL is 0, DML is the number of affected rows).
 	PrevAffectedRows int64
+	// AffectedRowsSetInForce is the affected rows set in a `set session_states` statement.
+	AffectedRowsSetInForce int64
```

Comment:
Why can not set `PrevAffectedRows` manually?

Reply:
Check out the code block here: https://github.com/pingcap/tidb/blob/50a2057dc96bda7192256fb11210eecb0d63dd57/executor/executor.go#L1947-L1954
- If the previous statement is INSERT/UPDATE/DELETE, the `AffectedRows` of the statement is assigned to the `PrevAffectedRows` of the following statement before the following statement begins. Otherwise, the `PrevAffectedRows` is 0 or -1.
- If the previous statement is `set session_states`, I also want to keep the `AffectedRows` so that `select row_count()` can get the affected rows.
- However, in the code above, I cannot judge the type of the previous statement because the previous statement is not passed in. One way is to add an `InSetSessionStateStmt` variable to `StmtCtx`, another is to add a `AffectedRowsSetInForce`.

Reply:
> Why can not set PrevAffectedRows manually?

`PrevAffectedRows` is assigned before a statement begins (in the code block I mentioned above). If I set `PrevAffectedRows` manually here, it won't be read anymore.

I do agree that adding an `AffectedRowsSetInForce` is strange. Any better ideas are welcome.

Reply:
> One way is to add an InSetSessionStateStmt variable to StmtCtx, another is to add a AffectedRowsSetInForce.

IMHO this is better than the `AffectedRowsSetInForce`(the semantic is clear).